File: lib/model/reffered_document/reffered_document.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'reffered_document.freezed.dart';
part 'reffered_document.g.dart';

@freezed
abstract class RefferedDocument with _$RefferedDocument {
  const factory RefferedDocument({
    required dynamic createdAt,
    required String refferedDocumentId,
    required String title,
    required String body,
    required String fileName,
    required String storageContentId,
    required String storageDocumentId,
    @Default(false) bool isDeleted,
  }) = _RefferedDocument;
  factory RefferedDocument.fromJson(Map<String, dynamic> json) =>
      _$RefferedDocumentFromJson(json);
}

File: functions/src/model/reffered_document.py
from pydantic import BaseModel
from datetime import datetime

class RefferedDocument(BaseModel):
    createdAt: datetime
    refferedDocumentId: str
    title: str
    body: str
    fileName: str
    storageContentId: str
    storageDocumentId: str
    isDeleted: bool
